### PR TITLE
flight_sql v1.1.19 (new formula)

### PR DESCRIPTION
**Pull request by prmoore77**
This adds the new Apache Arrow Flight SQL server example formula.  The formula installs the flight_sql server CLI executable.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/f/flight_sql.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    cd "build" do
+      system "cmake", "-S", "..", *std_cmake_args, "-G", "Ninja"
+      system "cmake", "--build", ".", "--target", "install"
+      bin.install "flight_sql"
+    end
+  end
+
+  test do
+    assert_equal "Flight SQL Server CLI: v1.1.19", shell_output("#{bin}/flight_sql --version").strip
+  end
```
Comment (bevanjkay) [inline Formula/f/flight_sql.rb:27]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?
Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Formula/f/flight_sql.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      tag:      "v1.1.19",
+      revision: "96f9831a901c4660c9cfb6cf299a682ef9614bb2"
+  license "Apache-2.0"
+
+  depends_on "boost" => :build
+  depends_on "cmake" => :build
+  depends_on "ninja" => :build
+  depends_on "openssl@3" => :build
+  depends_on xcode: :build
+  depends_on :macos
```
Comment (bevanjkay) [inline Formula/f/flight_sql.rb:14]: Upstream builds for Linux also, why is this necessary?

File: Formula/f/flight_sql.rb (hunk 1 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb:27]: ok - thanks, @bevanjkay - I'll build a better test...

File: Formula/f/flight_sql.rb (hunk 2 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb:14]: I got compiler errors on Mac for some reason using the default compilers that brew uses.  I was targeting Mac users with this formula, but wouldn't be opposed to building for both if I can get past the Mac compiler errors.  My upstream CI pipeline builds for MacOS fine, so I'm a little confused.

File: Formula/f/flight_sql.rb (hunk 1 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb:27]: This app runs a server until it is terminated manually by the user.  In the upstream Docker build CI (this [line](https://github.com/voltrondata/flight-sql-server-example/blob/96f9831a901c4660c9cfb6cf299a682ef9614bb2/Dockerfile.ci#L95) - I run this script: https://github.com/voltrondata/flight-sql-server-example/blob/main/scripts/test_flight_sql.sh - I can try to iterate on getting that to work in the brew formula, but I'm a bit over my head already :). I will try, however...

File: Formula/f/flight_sql.rb (hunk 2 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb:14]: if I remove the `depends_on: macos` line, would that be sufficient for supporting Linux as well?  I presume the Xcode dependency is ignored on Linux...

File: Formula/f/flight_sql.rb (hunk 1 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb]: I have updated the test to truly test the functionality of the server and client as requested.

**Comment by prmoore77** [comment]
Hi @bevanjkay - could you kindly re-review?  I believe I've addressed your concerns.  Thank you.

File: Formula/f/flight_sql.rb (hunk 2 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb:17]: I am disabling Linux build for now - I am getting cmake errors that are very difficult to understand.  This doesn't happen for brew Mac builds, and doesn't happen in my CI/CD pipeline where I build for Linux.  I'm targeting macOS users for this anyway - if ok, I would like to skip Linux for now.

**Comment by prmoore77** [comment]
@chenrui333 - could you review/approve, perhaps?  I would really like to get Flight SQL Server brew installable...  Thanks!

File: Formula/f/flight_sql.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  def install
+    mkdir "build"
+    cd "build" do
+      system "cmake", "-S", "..", *std_cmake_args, "-G", "Ninja"
+      system "cmake", "--build", ".", "--target", "install"
+      bin.install "flight_sql_server"
+      bin.install "flight_sql_client"
+    end
```
Comment (SMillerDev) [inline Formula/f/flight_sql.rb:26]: Please use the more modern syntax that defines the build directory for CMake. You can search for cmake in the repo for examples. There is also no need to set the build generator to ninja I suspect. Just let cmake pick

File: Formula/f/flight_sql.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      tag:      "v1.2.1",
+      revision: "6280bfb7642ac640f6352f01081a7a7b1805a318"
+  license "Apache-2.0"
+
+  depends_on "automake" => :build
+  depends_on "boost" => :build
+  depends_on "cmake" => :build
+  depends_on "gflags" => :build
+  depends_on "ninja" => :build
+  depends_on "openssl@3" => :build
```
Comment (SMillerDev) [inline Formula/f/flight_sql.rb:14]: I'm very surprised this is not a runtime dependency. Does it make a static build?

File: Formula/f/flight_sql.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  depends_on "python@3.11" => :build
```
Comment (SMillerDev) [inline Formula/f/flight_sql.rb:15]: Does it work with 3.12?

File: Formula/f/flight_sql.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  depends_on xcode: :build
+  depends_on :macos
```
Comment (SMillerDev) [inline Formula/f/flight_sql.rb:17]: I don't think we should define this instead of fixing the original issue.

Comment (prmoore77) [reply Formula/f/flight_sql.rb:17]: how does one troubleshoot linux builds with brew?  This works fine with my Github Actions pipeline, I'm perplexed as to why it fails with brew's testbot...

File: Formula/f/flight_sql.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/f/flight_sql.rb:14]: You shouldn't since that requires a recompile for security fixes

File: Formula/f/flight_sql.rb (hunk 6 above)
Comment (SMillerDev) [reply Formula/f/flight_sql.rb:17]: I'm not sure without seeing the error.

File: Formula/f/flight_sql.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  depends_on "openssl@3" => :build
+  depends_on "python@3.12" => :build
+  depends_on xcode: :build
+  depends_on :macos
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build", "--target", "install"
+    bin.install "flight_sql_server"
+    bin.install "flight_sql_client"
```
Comment (SMillerDev) [inline Formula/f/flight_sql.rb:23]: No `cmake --install`?

**Review by SMillerDev — COMMENTED** [review]
Reading through the build logs it seems to vendor DuckDB which would be against https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

File: Formula/f/flight_sql.rb (hunk 5 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb]: Yes - this is updated now.

File: Formula/f/flight_sql.rb (hunk 6 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb]: I got the Linux build to work, finally.  I've removed the "depends_on :macos" line...

File: Formula/f/flight_sql.rb (hunk 7 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb]: Fixed

File: Formula/f/flight_sql.rb (hunk 4 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb:14]: We prefer a static build to make it easy to install for non-brew users for the time being.  Does this prevent us from using brew as a distribution method?

**Comment by prmoore77** [comment]
> Reading through the build logs it seems to vendor DuckDB which would be against https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

We use headers from DuckDB's source that do not ship with the brew installation of DuckDB.

File: Formula/f/flight_sql.rb (hunk 3 above)
Comment (prmoore77) [reply Formula/f/flight_sql.rb]: Done - except for the Ninja part.  I found the build was faster with Ninja explicitly chosen...

**Comment by SMillerDev** [comment]
Could we adjust our formula to ship them?

**Comment by prmoore77** [comment]
> Could we adjust our formula to ship them?

I don't think they are intended to be public.  I'm using DuckDB internals as part of this product.  I usually have to change things upon major releases for that reason, but it is necessary to provide the functionality that this app needs.  They don't ship these headers as artifacts...

File: Formula/f/flight_sql.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/f/flight_sql.rb]: That doesn't matter for Homebrew since it will be a precompiled binary for most people. I do however prefer the reduced complexity of the default build system.

File: Formula/f/flight_sql.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/f/flight_sql.rb:14]: Yes, just like most other package managers we prefer dynamic linking since that will not require us to recompile all builds when there are openssl security updates.

Comment (prmoore77) [reply Formula/f/flight_sql.rb:14]: I understand, but I'm adopting a strategy similar to DuckDB which also statically links OpenSSL - how did they get approved?

Comment (SMillerDev) [reply Formula/f/flight_sql.rb:14]: DuckDB shouldn't have been added with static links to OpenSSL. But reviewers are human and can miss things.
